tt_metal/hw/inc/blackhole/noc_nonblocking_api.h

```diff
@@ -191,6 +207,32 @@ inline __attribute__((always_inline)) void noc_init(uint32_t atomic_ret_val) {
     }
 }
 
+inline __attribute__((always_inline)) void dynamic_noc_init() {
+#pragma GCC unroll 0
```

Comment:
this is good for now, but we did just move noc_init from an init routine to the performance path so we may consider unrolling at some point